src/ScheduleFunctions.cpp

```diff
@@ -461,17 +462,22 @@ Stmt build_produce(Function f) {
                     buf_name += ".buffer";
                     Expr buffer = Variable::make(type_of<struct buffer_t *>(), buf_name);
                     extern_call_args.push_back(buffer);
+                    buffer_names.push_back(buf_name);
                 }
             } else if (arg.is_buffer()) {
                 BufferPtr b = arg.buffer;
                 Parameter p(b.type(), true, b.dimensions(), b.name());
                 p.set_buffer(b);
-                Expr buf = Variable::make(type_of<struct buffer_t *>(), b.name() + ".buffer", p);
+                string buf_name = b.name() + ".buffer";
+                Expr buf = Variable::make(type_of<struct buffer_t *>(), buf_name, p);
                 extern_call_args.push_back(buf);
+                buffer_names.push_back(buf_name);
             } else if (arg.is_image_param()) {
                 Parameter p = arg.image_param;
-                Expr buf = Variable::make(type_of<struct buffer_t *>(), p.name() + ".buffer", p);
+                string buf_name = p.name() + ".buffer";
+                Expr buf = Variable::make(type_of<struct buffer_t *>(), buf_name, p);
                 extern_call_args.push_back(buf);
+                buffer_names.push_back(buf_name);
```

Comment:
Should we be annotating ImageParams passed to extern funcs? Shouldn't the calling program be doing that as appropriate? Note that this might explicitly mark data initialized that the host program actually failed to initialize, opening a "hole" in msan. 

The buffer_t object itself might need annotating though. To make the implementation not too crazy, it might be better to just store annotation Stmts in a vector, and insert them into the program below (where the Target::MSAN flag is checked now).